api/budget/api.go

```diff
@@ -173,6 +174,9 @@ func (c *client) doRequest(req interface{}, result interface{}) error {
 		}
 		resp, err = c.h.DoWithBody(req, bytes.NewReader(payload.Bytes()))
 		if err != nil {
+			if strings.HasSuffix(err.Error(), "Connection refused") {
```

Comment:
Is there a stronger test for this (perhaps by using juju/errors or errgo, whatever the return error is wrapped with), or is the underlying error completely masked/flattened?